prow/moonraker/moonraker.go

```diff
@@ -0,0 +1,83 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package moonraker
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/sirupsen/logrus"
+
+	prowapi "k8s.io/test-infra/prow/apis/prowjobs/v1"
+	"k8s.io/test-infra/prow/config"
+)
+
+const (
+	PathGetInrepoconfig = "inrepoconfig"
+)
+
+type Moonraker struct {
+	ConfigAgent       *config.Agent
+	InRepoConfigCache *config.InRepoConfigCache
+}
+
+type ProwYAMLGetter interface {
+	GetProwYAML(refs *prowapi.Refs) (*config.ProwYAML, error)
+}
+
+// serveGetInrepoconfig returns a ProwYAML object marshaled into JSON.
+func (mr *Moonraker) ServeGetInrepoconfig(w http.ResponseWriter, r *http.Request) {
```

Comment:
Are you planning to add unit tests for this or just rely on the integration tests?

Reply:
Resolving, per your answer at https://github.com/kubernetes/test-infra/pull/30294#discussion_r1330865467, which reads:

> Originally I was thinking neither the unit nor integration tests were properly validating that Moonraker returns the correct ProwJob YAML for different PRs. However, this appears to be well tested in https://github.com/kubernetes/test-infra/blob/master/prow/config/cache_test.go and Moonraker's serve function is a simple marshal/unmarshal wrapper so this integration test seems sufficient. 👍